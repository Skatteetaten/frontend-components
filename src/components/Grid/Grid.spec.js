import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Grid } from '.';

function oppsettFullDOM(props) {
  return mount(
    <Grid {...props}>
      <Grid.Row rowSpacing={Grid.SPACE_LARGE}>
        <Grid.Col lg={4}>4</Grid.Col>
        <Grid.Col lg={4} />
        <Grid.Col lg={4}>4</Grid.Col>
      </Grid.Row>
      <Grid.Row>
        <Grid.Col lg={4}>4</Grid.Col>
        <Grid.Col lg={4} />
        <Grid.Col lg={4}>4</Grid.Col>
      </Grid.Row>
    </Grid>
  );
}

describe('Grid komponent', () => {
  it('matcher snapshot ', () => {
    const wrapper = oppsettFullDOM();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renderer Grid med riktig markup ', () => {
    const wrapper = oppsettFullDOM({});

    expect(wrapper.exists('GridCol')).toEqual(true);
    expect(wrapper.exists('GridRow')).toEqual(true);
    expect(wrapper.find('GridRow').first().prop('rowSpacing')).toEqual('24px');
  });
});
