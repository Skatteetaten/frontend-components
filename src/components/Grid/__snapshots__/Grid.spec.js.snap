// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid komponent matcher snapshot  1`] = `
<Grid
  padding="0 8px"
>
  <div
    className="ske-Grid-106 "
  >
    <GridRow
      rowSpacing="24px"
    >
      <div
        className="ske-Grid-row-109 "
      >
        <GridCol
          lg={4}
        >
          <div
            className="ske-Grid-col-110 "
          >
            4
          </div>
        </GridCol>
        <GridCol
          lg={4}
        >
          <div
            className="ske-Grid-col-110 "
          />
        </GridCol>
        <GridCol
          lg={4}
        >
          <div
            className="ske-Grid-col-110 "
          >
            4
          </div>
        </GridCol>
      </div>
    </GridRow>
    <GridRow
      rowSpacing="8px"
    >
      <div
        className="ske-Grid-row-111 "
      >
        <GridCol
          lg={4}
        >
          <div
            className="ske-Grid-col-110 "
          >
            4
          </div>
        </GridCol>
        <GridCol
          lg={4}
        >
          <div
            className="ske-Grid-col-110 "
          />
        </GridCol>
        <GridCol
          lg={4}
        >
          <div
            className="ske-Grid-col-110 "
          >
            4
          </div>
        </GridCol>
      </div>
    </GridRow>
  </div>
</Grid>
`;
